File: python/nt_toolbox/plot_mesh.py
import numpy as np
import matplotlib.pyplot as plt
from skimage import measure
from mpl_toolkits.mplot3d import Axes3D
    
def plot_mesh(X, F, subplot = [1,1,1], title="", el=90, az=-90, lwdt=.1, dist=6, c="grey"):
    """
    plots the mesh defined by X : vertices and F : faces
    """
    ax = plt.subplot(subplot[0],subplot[1],subplot[2], projection='3d')
    ax.plot_trisurf(X[0,:], X[1,:], X[2,:], triangles=np.transpose(F), lw=lwdt, color=c, alpha=1)
    ax.axis("off")
    ax.view_init(elev=el, azim=az)
    ax.dist = dist
    plt.title(title)

File: julia/NtToolBox/src/plot_mesh.jl
function plot_mesh(X, F; sub = [1,1,1], titre="", el=90, az=-90, lwdt=.1, dist=6, c="grey")
    """
    plots the mesh defined by X : vertices and F : faces
    """
    subplot(sub[1],sub[2],sub[3], projection="3d")
    plot_trisurf(X[1,:], X[2,:], X[3,:], triangles=F'-1, lw=lwdt, color=c, alpha=1)
    axis("off")
    gca()[:view_init](el, az)
    gca()[:dist]=dist
    title(titre)
end
